src/FileSystem.gren

```diff
@@ -6,6 +6,7 @@ module FileSystem exposing
     , ReadableFileHandle
     , WriteableFileHandle
     , ReadWriteableFileHandle
+    , DirEntry(..)
```

Comment:
Feels like this should be an opaque type. But I needed to expose the constructors for the kernel js. @robinheghan any thoughts there?

Reply:
I think it's fine as a custom type, for now at least. I  don't expect these variants to change any time soon, and a new major release is expected to be released with Gren 0.3 this summer.

Reply:
My concern is losing a guarantee that, for example, `File "a.text"` is a file as determined by the low-level kernel js, and not some client code that manually created a `File "not-a-file"`. Maybe that's fine? Just gave me pause as this could  become more prevalent as more node apis are exposed. But if it's not a concern to you right now then this PR is ready to merge.

Reply:
Gotcha.

I think it’s fine for now.